ddl/placement/bundle.go

```diff
@@ -49,6 +49,224 @@ func NewBundle(id int64) *Bundle {
 	}
 }
 
+// NewBundleFromConstraintsOptions will transform constraints options into the bundle.
+func NewBundleFromConstraintsOptions(constraints string, leaderConstraints string, learnerConstraints string, followerConstraints string, voterConstraints string, followerCount uint64, voterCount uint64, learnerCount uint64) (*Bundle, error) {
```

Comment:
The parameters is so long... It's hard to maintain and easy to make mistakes

Reply:
I personally don't mind the long parameter, especially if it is more likely an internal API(exported though). `PlacementSettings` will be a good replacement that satisfy your preference, resolve it.